prog/weaveproxy/main.go

```diff
@@ -52,6 +52,16 @@ func main() {
 	Info.Println("weave proxy", version)
 	Info.Println("Command line arguments:", strings.Join(os.Args[1:], " "))
 
+	protoAddrParts := strings.SplitN(c.ListenAddr, "://", 2)
+	if len(protoAddrParts) == 2 {
+		c.ListenAddr = protoAddrParts[1]
+		if protoAddrParts[0] != "tcp" {
+			Error.Fatalf("Invalid protocol format: %q", protoAddrParts[0])
+		}
```

Comment:
nitpick: these two statements should be swapped, i.e. the check should come first.